src/Paprika/Store/DataPage.cs

```diff
@@ -308,4 +308,8 @@ public void Accept(IPageVisitor visitor, IPageResolver resolver, DbAddress addr)
             }
         }
     }
+    public Page Destroy(IBatchContext batch, in NibblePath prefix){
+        return new Page();
```

Comment:
`Destroy` should

1. COW
2. scan the map with `DeleteByPrefix`
3. delegate down to the specific bucket

A special case, if the prefix is empty, `RegisterPageForReuse` and delegate to children so they do it as well. COW properly.

Reply:
Hi @Scooletz , i have implemented the first 3 points for all Destroy methods. For RegisterForReuse, I can check if prefix.IsEmpty and then batch.RegisterForReuse(page). But how to delegate to children?

Reply:
If the prefix is empty, `batch.RegisterForReuse(page)` for this and propagate to all the children the `Destroy` call with the same empty prefix so that they can deal with it.